contrib/babelfishpg_tsql/sql/sys_functions.sql

```diff
@@ -74,6 +74,42 @@ END;
 $$
 LANGUAGE plpgsql IMMUTABLE;
 
+-- helper functions for XML EXIST(xpath)
+CREATE OR REPLACE FUNCTION sys.bbf_xmlexist(TEXT, ANYELEMENT)
+RETURNS sys.BIT
+AS
+$BODY$
+DECLARE
+    string_arg_datatype text;
+    string_basetype oid;
+    pltsql_quoted_identifier text;
+BEGIN
+    string_arg_datatype := sys.translate_pg_type_to_tsql(pg_typeof($2)::oid);
+    IF string_arg_datatype IS NULL THEN
+        -- for User Defined Datatype, use immediate base type to check for argument datatype validation
+        string_basetype := sys.bbf_get_immediate_base_type_of_UDT(pg_typeof($2)::oid);
+        string_arg_datatype := sys.translate_pg_type_to_tsql(string_basetype);
+    END IF;
+
+    IF (string_arg_datatype != 'xml') THEN
+        RAISE EXCEPTION 'Cannot call methods on %.', string_arg_datatype;
+    END IF;
+
+    pltsql_quoted_identifier := current_setting('babelfishpg_tsql.quoted_identifier');
+
+    IF (pltsql_quoted_identifier = 'off') THEN
+        RAISE EXCEPTION 'SELECT failed because the following SET options have incorrect settings: ''QUOTED_IDENTIFIER''. Verify that SET options are correct for XML data type methods.';
+    END IF;
+
+    IF xmlexists($1 passing by value $2) THEN
+        RETURN 1;
+    ELSE
+        RETURN 0;
+    END IF;
+END
+$BODY$
+LANGUAGE plpgsql IMMUTABLE STRICT PARALLEL SAFE;
```

Comment:
I think we shouldn't mark a function as IMMUTABLE if its output depends on the value of a

Reply:
Updated the volatility to STABLE